### PR TITLE
Merge & Access Policy

### PR DESCRIPTION
**Pull request by michaelcullum**
Merge policy

@php-fig/secretaries

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -17,3 +17,32 @@ license.
 By contributing documentation, examples, or any other non-code assets you agree
 to license your contribution under the CC BY 3.0 license. Attribution shall be
 given according to the current bylaws of this group.
+
+# Merge & Access Policy
+
+All Editors, Coordinators and Sponsors of specifications in draft stage have
```
Comment (samdark) [inline CONTRIBUTING.md]: Not only in draft stage, I guess.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -17,3 +17,32 @@ license.
...
+access to push to this repository. Secretaries have full administrative and
+push access. The master branch is protected and therefore cannot be forced
+pushed to.
+
+Guidelines for merging
```
Comment (samdark) [inline CONTRIBUTING.md]: Should be level 2 header `##`

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -17,3 +17,32 @@ license.
...
+* Never force push
```
Comment (samdark) [inline CONTRIBUTING.md]: To any branch or to master? Force-pushing to a pull request branch is typically OK.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -17,3 +17,32 @@ license.
...
+* All changes must go through pull requests, a commit should never be pushed
+directly
```
Comment (samdark) [inline CONTRIBUTING.md]: directly to master branch.

File: CONTRIBUTING.md (hunk 3 above)
Comment (samdark) [inline CONTRIBUTING.md]: If it's about master branch, it is not possible technically because branch protection.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -17,3 +17,32 @@ license.
...
+* All pull requests relating to a draft PSR must be approved (with a formal +1
+comment) or merged by the PSR Editor, except in the review phase when the coordinator
+should seek commment from the editor, but merging is at the coordinator's discretion
+* You must never merge a pull request that affects any file in the repository
+other than those you are on a working group for
```
Comment (samdark) [inline CONTRIBUTING.md]: Worth adding a note that secretary should be asked to do it

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -17,3 +17,32 @@ license.
...
+* You should never merge your own pull request
```
Comment (samdark) [inline CONTRIBUTING.md]: What if you're an editor?

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -17,3 +17,32 @@ license.
...
+* A change should never be merged to an accepted PSR without approval from
+secretaries, who will attempt to seek confirmation from the former spec Editors
+* A change to bylaws shouldn't be merged by anyone other than a secretary
+* Pull requests may be triaged (have labels applied) by anyone with push access,
+no matter which PSR they are on the working group for or which PSR it affects; but
+they cannot close a pull request or issue affecting other PSRs.
+
+These guidelines are subject to exceptions at secretaries discretion. Should
```
Comment (samdark) [inline CONTRIBUTING.md]: Better to say "changes" than "exceptions".

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -17,3 +17,32 @@ license.
...
+you have any questions please contact the secretaries on info [at] php-fig [dot] org.
+Failure to comply with guidelines will result in revokation of merge access. Merge
+access is a privillege and not a right and is simply granted to make your lives easier.
```
Comment (samdark) [inline CONTRIBUTING.md]: privillege → privilege

Comment (samdark) [inline CONTRIBUTING.md]: Last sentence could be removed. I see no purpose of it.

File: CONTRIBUTING.md (hunk 1 above)
Comment (michaelcullum) [inline CONTRIBUTING.md]: We remove access to the main repo after the draft/review stage but leave the groups intact for @mentions

File: CONTRIBUTING.md (hunk 6 above)
Comment (michaelcullum) [inline CONTRIBUTING.md]: Then you get your coordinator, sponsor or a secretary to merge it.

File: CONTRIBUTING.md (hunk 7 above)
Comment (michaelcullum) [inline CONTRIBUTING.md]: I mean exceptions.

**Comment by michaelcullum** [comment]
Working to setup https://pullapprove.com

**Comment by michaelcullum** [comment]
@php-fig Could you all please review this policy.

There is a sort of forced implementation of who can merge what on this repository by a tool called [pull approve](https://pullapprove.com/php-fig/fig-standards/). Essentially, it requires one of the WG to approve modifications to a PSRs file by them saying :+1: in the PR or hoping on the pull approve UI (Show all checks-> details-> click approve on yourself). There are certain directories (accepted, bylaws etc.) restricted to secretaries and the merge button is disabled when this approval has not occurred. Secretaries can override all of these approvals. You can see the config [here](https://github.com/php-fig/fig-standards/blob/master/.pullapprove.yml).

**Comment by michaelcullum** [comment]
@php-fig @php-fig/psr-5 @php-fig/psr-8 @php-fig/psr-9-10 @php-fig/psr-11 @php-fig/psr-12 @php-fig/psr-13 @php-fig/psr-14 @php-fig/psr-15 @php-fig/psr-16 @php-fig/psr-17 

Apologies for the mass mention, it seemed like an easier way to get your attention as I think most of you missed the mailing list email. Could you all please review this policy (and +1 the first post for acknowledgement).

There is a sort of forced implementation of who can merge what on this repository by a tool called [pull approve](https://pullapprove.com/php-fig/fig-standards/). Essentially, it requires one of the WG to approve modifications to a PSRs file by them saying :+1: in the PR or hoping on the pull approve UI (Show all checks-> details-> click approve on yourself). There are certain directories (accepted, bylaws etc.) restricted to secretaries and the merge button is disabled when this approval has not occurred. Secretaries can override all of these approvals. You can see the config [here](https://github.com/php-fig/fig-standards/blob/master/.pullapprove.yml).

**Pull request merged**